File: VoCal/VoCal/DevBypass.swift
//
//  DevBypass.swift
//  VoCal
//
//  Demo-mode bypass switches. When `enabled` is true, the app shows
//  "Skip" affordances in Onboarding and PaywallSheet that grant Pro
//  entitlement locally without going through Google/Apple sign-in or
//  StoreKit. Flip to `false` before App Store submission.
//

import Foundation

enum DevBypass {
    /// Controls visibility of demo-mode bypass buttons across the app.
    /// Read at view-render time, so flipping in code recompiles immediately.
    static let enabled: Bool = true

    /// Default profile applied when the user taps "Skip onboarding".
    /// Mid-range defaults so the app has sensible totals out of the box.
    static func defaultProfile() -> UserProfile {
        UserProfile(
            displayName: "Demo",
            streakDays: 0,
            weightLbs: 165,
            heightInches: 68,
            dailyCalorieGoal: 2200,
            sex: "",
            birthYear: 1995,
            entitlement: .pro
        )
    }

    /// Default daily calorie goal used by skip-onboarding bypass.
    static let defaultCalorieGoal: Int = 2200
}

File: vocal_flutter/lib/dev_bypass.dart
// Flip `enabled` to false before Play Store submission. When true, the
// app shows "Skip" buttons in Onboarding and PaywallSheet that grant
// Pro entitlement locally without auth or StoreKit. Mirrors the iOS
// DevBypass shim — same field names, same defaults.
//
// IMPORTANT: this is a demo / hackathon affordance, NOT a runtime feature
// flag. Toggling it requires a rebuild; there is no remote kill-switch.
// Production releases MUST ship with `enabled = false`.

import 'models/models.dart';

class DevBypass {
  /// Master switch — wraps every Skip CTA in the UI. Set false before
  /// submitting to the Play Store; the constness lets Dart's tree-shaker
  /// strip the Skip buttons from the release bundle.
  static const bool enabled = true;

  /// Default Pro profile used when the user taps Skip. Values mirror the
  /// iOS DevBypass.defaultProfile() so a demo built on either platform
  /// behaves the same — ~average male adult, 2200 kcal target.
  ///
  /// NOTE: UserProfile.heightInches is declared as double on this branch
  /// (see models.dart:212) — pass numeric literals as `.0` to avoid a
  /// silent int→double widening warning on stricter analyzer modes.
  static UserProfile defaultProfile() {
    return UserProfile(
      displayName: 'Demo',
      streakDays: 0,
      sex: '',
      heightInches: 68.0,
      weightLbs: 165.0,
      birthYear: 1995,
      dailyCalorieGoal: 2200,
      entitlement: Entitlement.pro,
    );
  }
}
